### DISCUSSION TITLE
How to include client flags in production?

### DISCUSSION THREAD

**Discussion by rgaino** [prisma/prisma#25539] [Q&A] [answered]
### Question

I added the `typedSql` feature to `previewFeatures` and, as explained in the docs, it needs the `--sql` added to `prisma generate` to install the `sql` module. But in production envs, the client is generated without flags.

Shouldn't this flag be respected if `typedSql` is listed under `previewFeatures`?

### How to reproduce

When deploying to a production env, and trying to use typed SQL:

```
node:internal/modules/cjs/loader:1225
  const err = new Error(message);
              ^

Error: Cannot find module '.prisma/client/sql'
Require stack:
- /app/node_modules/@prisma/client/sql.js
...
```

### Expected behavior

The `sql` module is installed with `prisma generate` in the `postinstall` hook because it's listed under `previewFeatures`.

### Information about Prisma Schema, Client Queries and Environment (optional)

My `schema.prisma`:
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["relationJoins", "prismaSchemaFolder", "typedSql"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

$ prisma -v
```
prisma                  : 5.21.1
@prisma/client          : 5.21.1
Computed binaryTarget   : debian-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v20.18.0
Query Engine (Node-API) : libquery-engine bf0e5e8a04cada8225617067eaa03d041e2bba36 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli bf0e5e8a04cada8225617067eaa03d041e2bba36 (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.21.1-1.bf0e5e8a04cada8225617067eaa03d041e2bba36
Default Engines Hash    : bf0e5e8a04cada8225617067eaa03d041e2bba36
Studio                  : 0.503.0
Preview Features        : relationJoins, prismaSchemaFolder, typedSql
```

**Comment by ludralph** [comment, marked as answer]
Hi @rgaino 👋 

> Shouldn't this flag be respected if typedSql is listed under previewFeatures?

The `typedSql` preview feature in your `schema.prisma` file enables the functionality, but the `--sql` flag is still required when running prisma generate to create the necessary TypeScript functions and types for SQL queries. This applies to both development and production environments.

To use TypedSQL in production:

1. Ensure a running database is accessible during client generation.
2. Modify your deployment process to include `--sql` flag (e.g., `prisma generate --sql`).

We are aware of this limitation and considering solutions to simplify the process in the future, as discussed in this [GitHub issue](https://github.com/prisma/prisma/issues/25124).
